### PR TITLE
tests/e2e: replace deprecated registry to use the k8s community

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
As part of the effort described here https://github.com/kubernetes/k8s.io/issues/1525 we are removing the mentions for any image that uses `gcr.io/gke-releases` and replacing to use the images located in the community registries.

the image `gcr.io/gke-release/gcp-compute-persistent-disk-csi-driver` is not available in the staging registry and in the prod registry

for example:
**Staging**: `skopeo inspect docker://gcr.io/k8s-staging-cloud-provider-gcp/gcp-compute-persistent-disk-csi-driver:v1.2.1 --raw`

Prod: `skopeo inspect docker://k8s.gcr.io/cloud-provider-gcp/gcp-compute-persistent-disk-csi-driver:v1.2.1 --raw`

PRs related to making that happen:
- https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/724
- https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/734
- https://github.com/kubernetes/test-infra/pull/21374
- https://github.com/kubernetes/k8s.io/pull/1854 **(Image promotion)**

/assign @spiffxp @msau42 @mattcary 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml [hunk 1]
```diff
@@ -56,7 +56,7 @@ spec:
             - name: socket-dir
               mountPath: /csi
         - name: gce-pd-driver
-          image: gcr.io/gke-release/gcp-compute-persistent-disk-csi-driver:v1.0.1-gke.0
```
Comment (mattcary) [inline test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml:59]: The sidecar versions above should be updated as well if we're updating the driver version. The versions used here should be good: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/deploy/kubernetes/images/stable-master/image.yaml#L1

**Review by mattcary — COMMENTED** [review]
/assign @saikat-royc 

Saikat, could you double check to see if anything else in the deployment needs to change with the update to 1.2.1? Especially with the daemonset (were we really skewed with 1.0 in the master and 0.7 on the daemonset)? Anyway I suggested to update the sidecar versions to match stable-master (which is still using the 1.2.1 driver but should be good enough?).

**Review by spiffxp — COMMENTED** [review]
/hold
/milestone v1.22
I would like to hold off on merging this until v1.21 is released

**Comment by msau42** [comment]
@saikat-royc can you help look into what else needs to be updated in the manifests? Looks like all the tests are failing after updating the sidecars.

**Comment by saikat-royc** [comment]
The test [log](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/100747/pull-kubernetes-e2e-gce-csi-serial/1378269347687960576/) are failing due to all the sidecars (provisioner, attacher, snapshotter, resizer, ) showing following error
```
W0403 11:23:00.192356       1 connection.go:172] Still connecting to unix:///csi/csi.sock
rpc error: code = NotFound desc = an error occurred when try to find container "ae89e4530bd8ce00ad5ee6919aa0781e1900f2f81fbae4df7e8ffe4cbb10398f": not found
```

the gce-pd-driver container is in a fatal loop:
```
==== start of pod log for container csi-gce-pd-controller-0/gce-pd-driver ====
F0403 11:17:23.120722       1 main.go:71] version must be set at compile time
```

I see that GKE internal prow test on k8s master are reporting similar csi.sock issue. However the same driver on GKE 1.16-1.19 is working fine.
still looking ...

**Comment by cpanato** [comment]
@saikat-royc @msau42 looking at the cloudbuild and job config we are setting `GCE_PD_CSI_STAGING_VERSION` that sets `STAGINGVERSION` in the makefile so I guess we are ok with that

**Comment by cpanato** [comment]
found the issue, fixing

**Comment by cpanato** [comment]
@msau42 @saikat-royc @mattcary PRs to fix the issue:

- main branch: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/761
- release-1.2 branch: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/760


we will need a release v1.2.2 to fix the image

**Comment by mattcary** [comment]
sg, I was about to push a 1.2.2 patch anyway.

**Comment by cpanato** [comment]
@mattcary @saikat-royc can you help me on the debugging why it is failing? I don't have much experience on the driver so I don't know if we need to update something else

**Comment by mattcary** [comment]
Very weird, the logs for this aren't showing up. I'll rerun the tests and see if we can see what's going on. It could be a flake, by the testgrid there has been some of that. https://testgrid.k8s.io/presubmits-kubernetes-nonblocking#pull-kubernetes-e2e-gce-storage-snapshot



/test pull-kubernetes-e2e-gce-storage-slow
/test pull-kubernetes-e2e-gce-storage-snapshot
/test pull-kubernetes-e2e-gce-csi-serial

**Comment by mattcary** [comment]
I feel like the pd csi driver is not starting well, but I'm having a hard time finding the right logs. I'd have to try to reproduce locally which will have to wait a few days.

**Comment by cpanato** [comment]
thanks @mattcary 
I will try to deploy it on my personal account.

**Comment by cpanato** [comment]
@mattcary @saikat-royc I don't have a GCP account that I can play around with to debug this, only my personal one which can become a bit expensive.

Can you, when having some free slots, help me with this? thanks in advance

**Comment by msau42** [comment]
The image should have been updated after https://github.com/kubernetes/kubernetes/pull/102212. I think all that's left now is to remove the registry.

**Comment by cpanato** [comment]
closing this was updated in https://github.com/kubernetes/kubernetes/pull/102212

thanks @mattcary 

/close

**Pull request closed without merging**